### PR TITLE
docs/contributing: add contributing guidelines doc and update readme

### PR DESCRIPTION
**Pull request by S-Coyle**
Add the new CONTRIBUTING.md document, and update the README to link to it

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# Contributing to the SAFE Network
+
+:tada: Thank you for your interest in contributing to the SAFE Network! :tada:
+
+This document is a set of guidelines for contributing to the SAFE Network. These are guidelines, not rules. This guide is designed to make it easy for you to get involved.
+
+As an open source project, we're excited to encourage and accept contributions from our community and beyond, and are striving to make that as easy and enjoyable as possible.
+
```
Comment (dirvine) [inline CONTRIBUTING.md]: that -> the process?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+Notice something amiss? Have an idea for a new feature? Feel free to create an Issue in this GitHub repository about anything that you feel could be fixed or improved. Examples include:
+- Bugs, crashes
+- Enhancement ideas
+- Unclear documentation
+- Lack of tutorials and hello world examples
+- ... and more
+
+See our [Issues and Feature Requests](#issues-and-feature-requests) section below for further information on creating new Issues.
+
+Of course, after submitting an Issue you are free to tackle the problem yourself, or pick up any of the other outstanding issues yet to be actioned - see the [Development](#development) section below for more information.
```
Comment (dirvine) [inline CONTRIBUTING.md]: Should e say, assign yourself to let us know you are on it and tackle it yourself?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+Further below we also have some information on where to go for [further help and support](#support).
```
Comment (dirvine) [inline CONTRIBUTING.md]: Further below we also have some information on where to go fo -> Further support is available here ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+This project adheres to the [Contributor Covenant](https://www.contributor-covenant.org/). By participating, you are expected to honour this code.
```
Comment (dirvine) [inline CONTRIBUTING.md]: you are expected -> we sincerely hope that you

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+## What we're working on
+
+The best way to follow our progress is to read the [MaidSafe Dev Updates](https://safenetforum.org/c/development/updates), which are published every week (on Thursdays) on the [SAFE Network Forum](https://safenetforum.org/).
+
+See our [Development Roadmap](https://safenetwork.tech/roadmap/) for more information on our near term development focus and longer term plans.
+
+## Issues and Feature Requests
```
Comment (dirvine) [inline CONTRIBUTING.md]: we need to remember to update this when we get the templates in place. Perhaps an issue somewhere

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+Fixes to review comments should preferably be pushed as additional commits to make it easier for the reviewer to see the changes. As a final step once the reviewer is happy the author should consider squashing these fixes with the relevant commit.
+
+Smaller PRs can have their commits squashed together and fast-forward merged, while larger PRs should probably have the chain of commits left intact and fast-forward merged into the upstream branch. This ensures that the chain of first parents in the main branch is checked by CI, and enables efficient use of `git bisect`.
+
+Where appropriate, commits should always contain tests for the code in question.
+
+#### Running tests (CI script)
+
+Submitted PRs are expected to pass continuous integration (CI), which, among other things, runs a test suite on your PR to make sure that your code doesn't have any bugs.
```
Comment (dirvine) [inline CONTRIBUTING.md]: doesn't have any bugs. -> has not regressed the code base.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+* **To do**: new issues which need to be reviewed and evaluated to decide their priority, add labels, clarify, etc
+* **In Progress**: the task is assigned to a person and it is in progress
+* **Needs Review**: the task is considered complete by the assigned developer and so has been sent for peer reviewed
+* **Reviewer approved**: the task has been approved by reviewer/s and is considered ready to be merged
+* **Done**: the PR associated with the task was merged (or the task was completed by any other means)
+
+The project board columns would typically include automation to move the issues between columns upon set actions, for example, if a PR was created which indicated in its description that it resolved a particular issue on the project board (using [GitHub keywords](https://help.github.com/articles/closing-issues-using-keywords)) then that issue would automatically be moved to the `Done` column on the board on PR merge.
+
+## Releases and Changelog
```
Comment (dirvine) [inline CONTRIBUTING.md]: I would still like to see us automate this via `cargo release`, that does not update the changelog, but we could make a changelog PR the trigger for cargo release?

Comment (dirvine) [reply CONTRIBUTING.md]: Probably should differentiate release of a lib and an application where we provide binaries.

Comment (S-Coyle) [reply CONTRIBUTING.md]: Agreed about the lib & binaries point - I'll update to include something to differentiate.
I didn't want to mention anything about `cargo` so as the guide could be generic enough to be added to all our repos, Rust or other

File: CONTRIBUTING.md (hunk 5 above)
Comment (S-Coyle) [reply CONTRIBUTING.md]: https://github.com/maidsafe/safenetwork.tech/issues/522 - have assigned this to me to update the roadmap

**Pull request merged**